src/payment/models/buy-crypto/entities/buy-crypto.entity.ts

```diff
@@ -142,13 +153,21 @@ export class BuyCrypto extends IEntity {
   }
 
   confirmSentMail(): this {
-    this.recipientMail = this.buy.user.userData.mail;
+    this.recipientMail = this.user.userData.mail;
     this.mailSendDate = Date.now();
 
     return this;
   }
 
-  get targetAddress(): string {
-    return this.buy.deposit ? this.buy.deposit.address : this.buy.user.address;
+  get user(): User {
+    return this.buy ? this.buy.user : this.cryptoRoute.user;
+  }
+
+  get target(): { address: string; asset: Asset } {
+    if (this.buy) {
+      return { address: this.buy.deposit ? this.buy.deposit.address : this.buy.user.address, asset: this.buy.asset };
+    } else {
+      return { address: this.cryptoRoute.targetDeposit.address, asset: this.cryptoRoute.asset };
```

Comment:
Need the same logic for cryptoRoute as for buy (only use targetDeposit address if there is one).